### PR TITLE
Mobile: fix geolocation

### PR DESCRIPTION
**Pull request by roman-r-m**
Fixes https://github.com/laurent22/joplin/issues/3240

### CODE REVIEW

**Comment by roman-r-m** [comment]
@tessus I don't think it's specific to android actually

**Comment by tessus** [comment]
@roman-r-m yes, you are right. I missed the podfile.

File: ReactNativeClient/android/app/build.gradle [hunk 1]
```diff
@@ -180,6 +180,7 @@ android {
 }
 
 dependencies {
+    implementation project(':@react-native-community_geolocation')
 	implementation "org.webkit:android-jsc:r241213"
```
Comment (laurent22) [inline ReactNativeClient/android/app/build.gradle:184]: As we're using RN 60, this shouldn't be necessary. Could you confirm if it works without it please?

File: ReactNativeClient/android/settings.gradle [hunk 2]
```diff
@@ -1,4 +1,6 @@
 rootProject.name = 'Joplin'
+include ':@react-native-community_geolocation'
+project(':@react-native-community_geolocation').projectDir = new File(rootProject.projectDir, '../node_modules/@react-native-community/geolocation/android')
```
Comment (laurent22) [inline ReactNativeClient/android/settings.gradle:3]: Same here, that should not be needed thanks to auto-linking.

File: ReactNativeClient/ios/Podfile [hunk 3]
```diff
@@ -32,6 +32,8 @@ target 'Joplin' do
   pod 'Folly', :podspec => '../node_modules/react-native/third-party-podspecs/Folly.podspec'
   pod 'rn-fetch-blob', :path => '../node_modules/rn-fetch-blob'
   use_native_modules!
+  pod 'react-native-geolocation', :path => '../node_modules/@react-native-community/geolocation'
+
```
Comment (laurent22) [inline ReactNativeClient/ios/Podfile:36]: Also should not be needed.

Comment (roman-r-m) [reply ReactNativeClient/ios/Podfile]: Removed

File: ReactNativeClient/android/settings.gradle (hunk 2 above)
Comment (roman-r-m) [reply ReactNativeClient/android/settings.gradle]: Removed

File: ReactNativeClient/android/app/build.gradle (hunk 1 above)
Comment (roman-r-m) [reply ReactNativeClient/android/app/build.gradle]: It does, removed it.

**Comment by roman-r-m** [comment]
I confirmed it works on Android but I've no way of testing on iOS

**Comment by laurent22** [comment]
I've merged the pull request in this commit: 0f4a781df06f6783feaab5acaac2b6bc1a4d8744 as I've also added the iOS config. I've set you as a co-author (it's in the commit description) but it doesn't look like GitHub picked it up.

In fact, I still haven't found out how to add my own commits to a GitHub pull request. I would think that's a common use case but it doesn't seem to be possible.

**Comment by tessus** [comment]
@laurent22 

> In fact, I still haven't found out how to add my own commits to a GitHub pull request. I would think that's a common use case but it doesn't seem to be possible.

You have to checkout the pull request, make changes, commit them, and push them.

You can checkout the PR manually or with one of the tools `hub` or `gh`.

**Pull request closed without merging**